spec/rubocop/cop/rails/has_many_or_has_one_dependent_spec.rb

```diff
@@ -49,5 +59,15 @@ class Person
     it 'does not register an offense when specifying `:dependent` strategy' do
       expect_no_offenses('has_many :foo, dependent: :bar')
     end
+
+    context 'with :through option' do
+      it 'does not register an offense for non-nil value' do
+        expect_no_offenses('has_many :foo, through: :bars')
+      end
+
+      it 'registers an offense for nil value' do
+        expect_no_offenses('has_many :foo, through: :nil')
```

Comment:
Where is this behaviour documented though? I've never seen `through: :nil`, can't figure out what it would do, nor find any documentation for it. 🤔